### ISSUE TITLE
Colors.xml: use two levels of color definition

### ISSUE THREAD

**Issue by felixwiemuth** [futurice/android-best-practices#177]
You suggest:
- Don't use `<color name="button_foreground">#FFFFFF</color>`
- Use `<color name="white">#FFFFFF</color>`

This makes it easier to change the colors used in the app but if you only declare color names and hard-code their usage you cannot change the foreground color of all buttons easily, for example.

My suggestion:
- Define color names in `colors.xml`, as you suggest
- Define color aliases for specific usage in separate resource files (or in `colors.xml` itself if appropriate).

Then you would end up with:
```
<color name="white">#FFFFFF</color>
...
<color name="button_foreground">@color/white</color>
...
```
In code you would then use the aliases.

**Comment by peter-tackage** [comment]
The current wording we have could be better explained by highlighting that it should be done in combination with styles.  So:
- `colors.xml` defines a pure color palette. (e.g. white is "#FFFFFF")
- `styles.xml` references the color palette and reflects the color usage. (e.g. the button foreground is white).
- `my_layout.xml` - references the style in `styles.xml` to color the button.

One could also add another layer of color abstraction between the `style.xml` and `color.xml` which defines the colors by their usage as you suggested. But it's a question of whether this brings more values than the complexity that it adds.

**Comment by felixwiemuth** [comment]
Exactly, for example I haven't noticed the usage of this pattern in section [Styles](https://github.com/futurice/android-best-practices#styles).

I think the additional layer is a good idea if you want to be flexible in changing the overall appearance of your app. Then you only have to change `button_foreground` whereas otherwise you would have to either change the actual color in `colors.xml` (possibly changing *too much*) or the color used in `my_layout.xml` (*too little* as you have to change it in each layout).

In general I think the depth and form of the hierarchy should be app-dependant while adhering to the basic concept described here.

**Comment by peter-tackage** [comment]
I'll update the section to give a more complete view of the options.

**Comment by peter-tackage** [comment]
Resolved in #178
